ARCs/arc-0063.md

```diff
@@ -0,0 +1,320 @@
+---
+arc: 63
+title: Lsig Plug-In Signer for Msig Vault
+description: Delegated multisig-account controlled by one account
+author: Stéphane BARROSO (@SudoWeezy)
+discussions-to: https://github.com/algorandfoundation/ARCs/issues/303
+status: Draft
+type: Standards Track
+category: ARC
+created: 2024-07-16
+---
+
+## Abstract
+
+This ARC proposes a method for creating a delegated multisig account controlled by one account and a Logic Signature (Lsig).
+
+## Motivation
+
+The motivation behind this ARC is to extend Algorand account features by enabling third-party "Plug-Ins" using a combination of delegated Lsig and Multi-Signature accounts, which act as vaults. This approach allows anyone to sign the Lsig for the vault, while maintaining security and control through a classic algorand account.
+
+## Specification
+
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in <a href = "https://www.ietf.org/rfc/rfc2119.txt">RFC-2119</a>
+
+### Components
+
+1. **Owner Account**: Vault's Owner
+1. **Lsig Plug-In**: Provided by a third party.
+1. **Plug-In Signer**: Created by generating a new key pair
+1. **1/2 Msig Account**: Comprises the owner's address, and the plug-in signer.
+
+### Implementation 1
+
+We will use the following Logic plug-in for illustrative purposes:
+
+**DO NOT USE IN PRODUCTION**
+
+```python
+teal_program = """
+#pragma version 10
+txn TypeEnum
+pushint 4
+==
+txn AssetAmount
+pushint 0
+==
+&&
+txn RekeyTo
+global ZeroAddress
+==
```

Comment:
```suggestion
==
&&
txn AssetCloseTo
global ZeroAddress
==
```

Reply:
Good catch, to be even more strict, we could also add this: 
```
==
global GenesisHash
byte base64(wGHE2Pwdvd7S12BL5FaOP20EGYesN73ktiC1qzkkit8=)
==
&&
==
&&
gtxn 1 FreezeAssetAccount
global ZeroAddress
==
```